docker/run_lints_in_docker.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# Script that sets up the docker environment to run the lints in and runs the
+# lints.
+
+# Failures should cause setup to fail
+set -v -e -x
+
+DC="$(which docker-compose)"
+APP_UID="10001"
+APP_GID="10001"
+
+# Use the same image we use for building docker images because it's cached.
+# Otherwise this doesn't make any difference.
+BASEIMAGENAME="python:2.7.14-slim"
+
+# Figure out which image to run lints in
+LINTIMAGE="local/socorro_webapp"
+
+# Create a data container to hold the repo directory contents and copy the
+# contents into it
+if [ "$(docker container ls --all | grep socorro-repo)" == "" ]; then
+echo "Creating socorro-repo container..."
+docker create \
+        -v /app \
+        --user "${APP_UID}" \
+        --name socorro-repo \
+        "${BASEIMAGENAME}" /bin/true
+fi
+echo "Copying contents..."
+# Wipe whatever might be in there from past runs
+docker run \
+        --user root \
+        --volumes-from socorro-repo \
+        --workdir /app \
+        "${LINTIMAGE}" sh -c "rm -rf /app/*"
+
+# Verify files are gone
+docker run \
+        --user root \
+        --volumes-from socorro-repo \
+        --workdir /app \
+        "${LINTIMAGE}" ls -l /app/
+
+# Copy the repo root into /app
+docker cp . socorro-repo:/app
+
+# Fix permissions in data container
+docker run \
+        --user root \
+        --volumes-from socorro-repo \
+        --workdir /app \
+        "${LINTIMAGE}" chown -R "${APP_UID}:${APP_GID}" /app
+
+# Run cmd in that environment and then remove the container
+echo "Running lints..."
+docker run \
+        --rm \
+        --user "${APP_UID}" \
+        --volumes-from socorro-repo \
+        --workdir /app/webapp-django \
+        --tty \
+        --interactive \
+        "${LINTIMAGE}" /webapp-frontend-deps/node_modules/.bin/eslint .
+
+echo "Done!"
```

Comment:
I can see why you thought this was the right way to go, because we run flake8 in the same step we run pytest. That script is set up the way it is because pytest needs to run with other services running and we can't do that in Circle and mount volumes. So we have to create a data container and blah blah blah. The fact that flake8 runs via `run_tests.sh` isn't necessary. It'd be easier if we extricated it from `run_tests.sh`.

I think we have two possibilities here:

1. run eslint in `run_tests.sh` with everything else continuing what we've got now, or
2. run it as a separate line in `.circleci/config.yml`:
   ```
   docker run local/socorro_webapp /webapp-frontend/deps/node_modules/.bin/eslint
   ```
   We do this in Antenna currently:
   
   https://github.com/mozilla-services/antenna/blob/a700c68f288b85bb112068309c3b791f879629fd/.circleci/config.yml#L40
   
   We could even make a new make rule for this like we have for flake8 and then run that.

It's possible the first option isn't possible. I didn't look very closely.

Sorry for the confusion!

Reply:
I transferred the JS linting to `run_tests_in_docker.sh`, since I am not sure on how to access `/webapp-frontend/deps/node_modules/.bin/eslint` in `run_tests.sh` to run `eslint`.

I'm not sure only putting it in `circleci/config.yml` would be super convenient for development, since then you would not be able to run the linting locally.

Is having the JS linting in `run_tests_in_docker.sh` fine?

Reply:
I think my attempt to explain what we've currently got and why it's so weird overcomplicated things and made it harder to communicate what I'm thinking well. Sorry about that!

Running eslint should be really straight-forward because it's just running a command in a container and doesn't have the complications that running pytest has.

If you look at the "lint" rule in the `Makefile`, that's essentially all we have to do to run eslint. The difference is that we run `flake8` in the processor container and we need to run eslint in the `webapp` container.

When we run commands in a container, the command is run in the file-system context of the container--not your host machine. In the container, we have a `/webapp-frontend/deps/node_modules/.bin/eslint` binary. You can verify this:

```
$ docker-compose run webapp ls /webapp-frontend/deps/node_modules/.bin/
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                            this is run inside the container
```

You can run it like this:

```
$ docker-compose run webapp /webapp-frontend/deps/node_modules/.bin/eslint
```

I think adding an "eslint" rule to the `Makefile` and then calling that from Circle's config file seems like a good way to go here.

Does that make more sense? If not, we should switch to a video conference or something.